daffodil-cli/src/it/scala/org/apache/daffodil/blob/TestBlob.scala

```diff
@@ -0,0 +1,191 @@
+  val output2 = Util.getExpectedString("output2.txt")
+  val output4 = Util.getExpectedString("output4.txt")
+  val output6 = Util.getExpectedString("output6.txt")
+  val output8 = Util.getExpectedString("output8.txt")
+  val output9 = Util.getExpectedString("output9.txt")
+  val output10 = Util.getExpectedString("output10.txt")
+  val output12 = Util.getExpectedString("output12.txt")
+
+  // Call md5sum with two different files and return if they match
+  /*def md5sum(shell: Expect, a: String, b: String): Boolean {
+    val cmdA = String.format("/usr/bin/md5sum %s | awk '{print $1}'", a)
+    shell.sendLine(cmd)
+    val resA = shell.expect("\n").toString
+
+    val cmdB = String.format("/usr/bin/md5sum %s | awk '{print $1}'", b)
+    shell.sendLine(cmd)
+    val resB = shell.expect("\n").toString
+
+    return resA == resB
+  }*/
+
+  @Test def test_basic_blob() {
+
+    val schemaFile = Util.daffodilPath("daffodil-cli/src/it/resources/org/apache/daffodil/CLI/blob.dfdl.xsd")
+    val inputFile = Util.daffodilPath("daffodil-cli/src/it/resources/org/apache/daffodil/CLI/input/blob_01.bin")
+    val (testSchemaFile, testInputFile) = if (Util.isWindows) (Util.cmdConvert(schemaFile), Util.cmdConvert(inputFile)) else (schemaFile, inputFile)
+
+    val shell = Util.start("")
+
+    try {
+      val cmd = String.format("%s parse -s %s %s", Util.binPath, testSchemaFile, testInputFile)
+      shell.sendLine(cmd)
+      val result = shell.expect(contains("tns:blob_01")).toString
+      // Use +7 to drop the 'file://' to get the path of the file
+      val generated_blob = result.substring(result.indexOf("file://") + 7).takeWhile(_ != '<')
+
+      //val checksum = String.format("%s parse -s %s %s", Util.binPath, testSchemaFile, testInputFile)
+      val cmp = String.format("/usr/bin/cmp %s %s", generated_blob, testInputFile)
+      shell.sendLine(cmp)
+      shell.expect(contains("\n"))
+      shell.sendLine("echo $?")
+      shell.expect(contains("0\n"))
+
+      // Clean up blobs
+      shell.sendLine("rm -rf daffodil-blobs")
+
+      shell.send("exit\n")
+      shell.expect(eof)
+      shell.close()
+    } finally {
+      shell.close()
+    }
+  }
+
+  @Test def test_1GB_blob() {
+
+    val DAFFODIL_JAVA_OPTS = Map("DAFFODIL_JAVA_OPTS" -> "-Xms256m -Xmx512m")
+    val schemaFile = Util.daffodilPath("daffodil-cli/src/it/resources/org/apache/daffodil/CLI/1GB_blob.dfdl.xsd")
+    val inputFile = Util.daffodilPath("daffodil-cli/src/it/resources/org/apache/daffodil/CLI/input/1GB_blob.bin")
+    val (testSchemaFile, testInputFile) = if (Util.isWindows) (Util.cmdConvert(schemaFile), Util.cmdConvert(inputFile)) else (schemaFile, inputFile)
+
+    //val shell = Util.startIncludeErrors("", envp = DAFFODIL_JAVA_OPTS)
+    val shell = Util.startIncludeErrors("", envp = DAFFODIL_JAVA_OPTS)
+
+    try {
+      // Create 1GB blob file
+      val dd = String.format("/usr/bin/dd if=/dev/zero of=%s bs=1M count=1000", testInputFile)
```

Comment:
We can feed 1G+ of data to a pipe using only smallish buffers using Scala/Java code. 
Thread 1 writes into a PipedOutputStream, and main thread's daffodil's input stream is created from the corresponding PipedInputStream, and reads from it as normal.

Reply:
That's still going to result in a blob files being created since we only support file based blobs. It it does at least avoid haing the input files having to be on disk. Definiately a better route to go. Also, it doesn't seem too unreasonable to require 1GB+ of free space when running integration tests. This are mostly only ever run on CI machines where that shouldn't be a problem as long as we make sure to delete the blobs after a test finishes.

Reply:
I am coming around to your view that tests that push the real boundaries of this stuff should be manual test steps that we just run on specific machines. E.g., I think we need to do a 16GB test on a 4GB JVM that parses data including huge blob, then unparses, I also think we need  a test where that same huge blob comes out of a gzipped layer, so the blob, while huge, can come from a tiny test file (assuming the file is for example 16GB of the same 8 bytes 0xDEADBEEF)
I want to be sure there is nothing finite and problematic in the layering stuff. I fear there may be, though it mostly just reuses the I/O layer, so there's a chance this just composes.